### PR TITLE
Replace static fee system by dynamic fee system

### PR DESCRIPTION
**Pull request by IkerAlus**
Add the LIP "Replace static fee system by dynamic fee system".

****

Changes from the version found at _research.lisk_ forum:

- Minor format fixes.
- Fix the size of Delegate Registration transaction in Appendix.
- Add `Include fee property in transaction JSON object` subsection at Specification section.
- Change 'trs.fee' property format to big-endian.
- Include comment at Wallets subsection of Rationale section to address feedback from _research.lisk_ forum.

### CODE REVIEW

File: proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+With the assumption of having these “anti-spam” measures in the previously mentioned modules, it would be interesting to consider how these resources mentioned above are used by all the transaction types and the cost implied. Unfortunately, the calculation of the exact cost of these resources is very complex and depends on several unrelated points as, hardware specifications, connection bandwidth, software platform, cost of electricity, Lisk token volatility, etc. An accurate research about this would be extremely complex and it is out of the scope of this proposal. However, it is still important to study the price of the different transactions and the price of broadcasting several of them depending on the value of `minFeePerByte`:  
+
+- [Appendix A](#Appendix-A:-Minimum-fee-implications) has an overview of minimum transaction fees depending on different values of `minFeePerByte` and the estimated direct implications for the network.
```
Comment (AndreasKendziorra) [inline proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: The link seems to be incorrect. Also for the other occurrences of appendix links.

File: proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+### B: Namespace fee implications
+
+Minimum fee, `trs.minFee`, for transaction types 2 and 5 for different values of `trs.nameFee` for the current protocol reference implementation:
+
+1. **`trs.nameFee` = 0:**
+
+**Tx type/ Min fee** | **1*10<sup>-7</sup> LSK/byte** | **5*10<sup>-7</sup> LSK/byte** | **1*10<sup>-6</sup> LSK/byte** | **5*10<sup>-6</sup> LSK/byte** | **1*10<sup>-5</sup> LSK/byte**
+--- | --- | --- | --- | --- | --- 
+**Delegate reg (145 bytes)** | 1.45*10<sup>-5</sup> LSK | 7.25*10<sup>-5</sup> LSK | 1.45*10<sup>-4</sup> LSK | 7.25*10<sup>-4</sup> LSK | 0.0013 LSK
```
Comment (AndreasKendziorra) [inline proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: the last value must be 0.00145 LSK.
Also for the following tables, the last entry of the first row has to be updated.

File: proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+- **Include fee property in transaction signature and transaction ID**: To prevent transaction fee forging and manipulation by malicious nodes/delegates, the transaction fee has to be part of the byte array used as input for the transaction signature and transaction ID generation (e.g. implemented by [`getBytes`](https://github.com/LiskHQ/lisk/blob/5d5af525eec7f2ee687ca921a6a2d1457c471da9/logic/transaction.js#L154) in Lisk core). The `trs.fee` property, with a size of 8 bytes and encoded in big-endian, has to follow the `trs.senderPublicKey` property during serialisation. Additionally, the [`getTransactionBytes`](https://github.com/LiskHQ/lisk-elements/blob/d5ab830ad65965764d364eb344ce4e3a13bf14c9/packages/lisk-transactions/src/utils/get_transaction_bytes.js#L147) function must be implemented in the same way in Lisk Elements.
+
```
Comment (janhack) [inline proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: I would say `as input for the transaction signature generation` or `as input for the transaction signing`.

Comment (IkerAlus) [reply proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: changed to the first option

File: proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md (hunk 2 above)
Comment (IkerAlus) [reply proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: good catch! updated

File: proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md (hunk 1 above)
Comment (IkerAlus) [reply proposals/lip-replace_static_fee_system_by_dynamic_fee_system.md]: corrected all the appendix links

**Pull request merged**